### DISCUSSION TITLE
Fixed row as fixed column does

### DISCUSSION THREAD

**Discussion by JdeJ** [komarovalexander/ka-table#175] [Q&A] [answered]
Hi, first of all congrats for the library, it helps me a lot.

I have a table component with virtualization only on the body  and I have a row that gets fixed on top of the table body even when scroll down.

I have the header row also sticky but when I scroll down in the virtualized table, the header keeps on but the fixed row disappears.

The row y need to fix is always the first row.

Is there any way to do that with your table component? 

Thanks so much for your work

**Comment by komarovalexander** [comment, marked as answer]
Hi @JdeJ 
Thanks for the interesting question:
depends on the requirements but you can do it way like this
code: 
https://stackblitz.com/edit/table-many-rows-ts-z7nurn?file=Demo.tsx
```

        childComponents={{
          tableBody: {
            content: props => {
              const firstRowData = props.data.shift();
              const rowKeyValue = props.data[0].id;
              return (
                <>
                  <DataRow
                    {...props}
                    childComponents={{
                      dataRow: {
                        elementAttributes: () => ({ className: 'fixed-row' })
                      }
                    }}
                    rowData={firstRowData}
                    rowEditableCells={[]}
                  />
                  <TableBodyContent {...props} />
                </>
              );
            }
          }
        }}
```


styles:
https://stackblitz.com/edit/table-many-rows-ts-z7nurn?file=style.scss
```
.ka-table .fixed-row {
  top: 48px;
  position: sticky;
  background-color: white;
}
```

**Reply by JdeJ** [reply]
Wow, thanks for the response, it works like a charm.

**Comment by paukyduckyKore** [comment]
Thank you I got this to work as well with for the bottom summary row. Using this:

`.ka-table .fixed-row {
  bottom: 0px;
  position: sticky;
  background-color: white;
}`
